trunk/NDHMS/CPL/WRF_cpl/module_wrf_HYDRO.F

```diff
@@ -98,15 +100,15 @@ subroutine wrf_cpl_HYDRO(HYDRO_dt,grid,its,ite,jts,jte)
 
   
         if(.not. RT_DOMAIN(did)%initialized) then
-           
-           call MPP_LAND_INIT()
-
            !yw nlst_rt(did)%nsoil = config_flags%num_soil_layers
            !nlst_rt(did)%nsoil = model_config_rec%num_metgrid_soil_levels
            nlst(did)%nsoil = grid%num_soil_layers
 
          
 #ifdef MPP_LAND
+           call MPI_COMM_DUP(MPI_COMM_WORLD, HYDRO_COMM_WORLD, ierr)
+           call MPP_LAND_INIT(grid%e_we - grid%s_we - 1, grid%e_sn - grid%s_sn - 1)
```

Comment:
NOTE: These counts are the velocity points, so they need to be reduced by 1 for the actual surface cell counts.